addons/account/models/account_payment.py

```diff
@@ -890,16 +892,9 @@ def _create_paired_internal_transfer_payment(self):
             })
             paired_payment.move_id._post(soft=False)
             payment.paired_internal_transfer_payment_id = paired_payment
-
-            body = _(
-                "This payment has been created from %s",
-                payment._get_html_link(),
-            )
+            body = escape(_("This payment has been created from:")) + payment._get_html_link()
```

Comment:
it was multiline to show the link is inside he `_( )` call but this is no longer the case

Reply:
A lot of the changes are now harder to read though...

Why isn't the translation function escaping implicitly, instead of escaping it explicitly everywhere?
Maybe with a named only param for cases where it is annoying to change
```python
body = _(
    "This payment has been created from <b>%s</b>",
    payment._get_html_link(),
    _escape=False
)
```

AFAIK, translations can be an issue if they contain HTML in the source anyways (security and consistency in translation), so we probably don't even need that param.

Reply:
> Why isn't the translation function escaping implicitly, instead of escaping it explicitly everywhere?

Because the gettext calls are not always targetting HTML output. A lot of them are for text field and you risk of double escaping of valid content if you always escape.
Also the point of this PR is to remove the `message_post` ambiguity, not to change the translation framework.